openapi3/datetime_schema_test.go

```diff
@@ -0,0 +1,107 @@
+package openapi3
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var DateSpec = []byte(`
+components:
+  schemas:
+    Server:
+      properties:
+        date:
+          $ref: "#/components/schemas/timestamp"
+        name:
+          type: string
+      type: object
+    timestamp:
+      type: string
+      format: date
+openapi: "3.0.1"
+paths: {}
+info:
+  version: 1.1.1
+  title: title
+`[1:])
+
+var DateTimeSpec = []byte(`
+components:
+  schemas:
+    Server:
+      properties:
+        datetime:
+          $ref: "#/components/schemas/timestamp"
+        name:
+          type: string
+      type: object
+    timestamp:
+      type: string
+      format: date-time
+openapi: "3.0.1"
+paths: {}
+info:
+  version: 1.1.1
+  title: title
+`[1:])
+
+func TestDateZeroMonth(t *testing.T) {
+	loader := NewLoader()
+	doc, err := loader.LoadFromData(DateSpec)
+	require.NoError(t, err)
+
+	err = doc.Validate(loader.Context)
+	require.NoError(t, err)
+
+	err = doc.Components.Schemas["Server"].Value.VisitJSON(map[string]any{
+		"name": "kin-openapi",
+		"date": "2001-00-03",
+	})
+	require.ErrorContains(t, err, `Error at "/date": string doesn't match the format "date": string doesn't match pattern "`+FormatOfStringDate+`"`)
```

Comment:
Are you looking for EqualError?

Because otherwise, this could be enough

```suggestion
	require.ErrorContains(t, err, `Error at "/date": string doesn't match the format "date"`)
```

What do you think?

Reply:
👍